doc/article/en-US/templating-html-behaviors-introduction.md

```diff
@@ -297,7 +297,232 @@ The `@bindable` decorator isn't used when doing single value binding with a Cust
+
+<code-listing heading="extend-coerce">
+  <source-code lang="ES 2015">
+    bindable.usePropertyType(true);
+  </source-code>
+</code-listing>
+
+After doing this, all of the type in metadata emitted by Typescript will be used to resolve to its equivalent property in `coerceFunctions` above. By default, there are 4 mappings:
+
+  * `Number` to `'number'`
+  * `String` to `'string'`
+  * `Boolean` to `'boolean'`
+  * `Date` to `'date'`
+
+You can extend / modify this via export `coerceFunctionMap` of `aurelia-binding`:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="TypeScript">
+    // import from 'aurelia-binding' if you are writing a plugin
+    // import from 'aurelia-framework' if you are writing an application
+    import {coerceFunctions, coerceFunctionMap} from 'aurelia-framework';
+
+    // Add `point` coerce function
+    coerceFunctions.point = function(incomingValue) {
+      return incomingValue.split(' ').map(parseFloat);
+    };
+
+    // Register a class to be used with coerce
+    coerceFunctionMap.set(Point, 'point');
+
+    // then we can simply have
+    export class Line {
+      @bindable point1: Point
+      @bindable point2: Point
+    }
+  </source-code>
+</code-listing>
+
+In above example, `bindable` decorator auto picks up the type, set proper coerce function for `point1` and `point2`.
+
+JavaScript user can also leverage this pattern via `Reflect.metadata` decorator by decorating property with correct metadata for `propertyType` before decorating it with `bindable`, for example:
+
+<code-listing heading="extend-coerce">
+  <source-code lang="ES 2015">
+    // import from 'aurelia-binding' if you are writing a plugin
+    // import from 'aurelia-framework' if you are writing an application
+    import {coerceFunctions, coerceFunctionMap} from 'aurelia-framework';
+    import {metadata} from 'aurelia-metadata';
+
+    // Add `point` coerce function
+    coerceFunctions.point = function(incomingValue) {
+      return incomingValue.split(' ').map(parseFloat);
+    };
+
+    // Register a class to be used with coerce
+    coerceFunctionMap.set(Point, 'point');
+
+    // then we can simply have
+    export class Line {
+      @bindable
+      @Reflect.metadata(metadata.propertyType, Point)
+      point1
+      
+      @bindable
+      @Reflect.metadata(metadata.propertyType, Point)
+      point2
+    }
+  </source-code>
+</code-listing>
+
+In above example, `bindable` decorator auto picks up the type, set proper coerce function for `point1` and `point2`.
+
+#### Fluent syntax
+
+If you scroll back top a bit, you will notice we had fluent syntax decorator: `@bindable.number`. This, as described above, is a simplified and more expressive form of `coerce: 'number'`. Aurelia also provides a way to build your own fluent syntax `bindable` decorator. You can do this by:
```

Comment:
Perhaps instead of 
"If you scroll back **to**p a bit"
following would be better:
"If you scroll back **u**p a bit"